Permission check lost when creating folders

#### Software and hardware versions 
Pixel XL API 28 (Andrloid 9.0)
Nexus 6P API 23 (Android 6.0)

master branch
#### Problem description
App crashed when we didn't give it the permission to write the storage, see `createODKDirs()`. and we do not have ODK Collect before.

#### Steps to reproduce the problem
- install this app in your device first (without ODK Collect)
- make sure it has no permission gained
- the first time you launch this app, crash

#### Expected behavior
Add a permission checks 

#### Other information 
```
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.odk.share/org.odk.share.activities.MainActivity}: java.lang.RuntimeException: Cannot create directory: /storage/emulated/0/share
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
        at android.app.ActivityThread.-wrap11(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
     Caused by: java.lang.RuntimeException: Cannot create directory: /storage/emulated/0/share
        at org.odk.share.application.Share.createODKDirs(Share.java:56)
        at org.odk.share.activities.MainActivity.onCreate(MainActivity.java:84)
        at android.app.Activity.performCreate(Activity.java:6237)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
```

@opendatakit-bot claim

--- a/skunkworks_crow/src/main/java/org/odk/share/activities/MainActivity.java
+++ b/skunkworks_crow/src/main/java/org/odk/share/activities/MainActivity.java
@@ -81,7 +81,7 @@ public class MainActivity extends FormListActivity implements LoaderManager.Load
 
         setTitle(getString(R.string.app_name));
         setSupportActionBar(toolbar);
-        Share.createODKDirs(this);
+        createODKDirs();
 
         sendForms.setEnabled(false);
 
@@ -220,6 +220,17 @@ public class MainActivity extends FormListActivity implements LoaderManager.Load
 
         builder.setCancelable(false);
         builder.show();
+    }
+
+    // call createODKDirs() with a permission check.
+    private void createODKDirs() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M ||
+                checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
+            Share.createODKDirs(this);
+        } else {
+            requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE},
+                    STORAGE_PERMISSION_REQUEST_CODE);
+        }
     }
 
     private void addListItemDivider() {

